jboss-eap-config-jgroups/added/launch/ha.sh

```diff
@@ -86,6 +93,53 @@ get_socket_binding_for_ping() {
         echo "socket-binding=\"jgroups-mping\""
     fi
 }
+
+generate_jgroups_auth_config() {
+
+  local cluster_password="${1}"
+  local digest_algorithm="${2}"
+  local config
+
+  if [ -z "${cluster_password}" ]; then
+      log_warning "No password defined for JGroups cluster. AUTH protocol will be disabled. Please define JGROUPS_CLUSTER_PASSWORD."
+      config="<!--WARNING: No password defined for JGroups cluster. AUTH protocol has been disabled. Please define JGROUPS_CLUSTER_PASSWORD. -->"
+  else
+      config="\n <auth-protocol type=\"AUTH\">\n\
+                    <digest-token algorithm=\"${digest_algorithm:-SHA-512}\">\n\
+                        <shared-secret-reference clear-text=\"${cluster_password}\"/>\n\
+                    </digest-token>\n\
+                </auth-protocol>\n"
+  fi
+  echo ${config}
+}
+
+generate_ping_config() {
+    local ping_protocol="${1}"
+    local service_port="${2}"
+    local service_name="${3}"
+    local ping_namespace="${4}"
+    local socket_binding="${5}"
+    local config
+    # for DNS_PING, the record is my-port-name._my-port-protocol.my-svc.my-namespace
+    config="<protocol type=\"${ping_protocol}\" ${socket_binding}>"
+    if [ "${ping_protocol}" = "dns.DNS_PING" ]; then
+        if [ -z "${KUBERNETES_SERVICE_PORT_DNS}" ]; then
+            log_error "KUBERNETES_SERVICE_PORT_DNS is required for dns.DNS_PING."
+        fi
+        kdns_port=${KUBERNETES_SERVICE_PORT_DNS:-53}
+        if [ -z "${KUBERNETES_SERVICE_PORT_DNS}" ]; then
+            log_warn "KUBERNETES_SERVICE_PORT_DNS is not set, using default value of 53."
```

Comment:
Is 53 the typical value?  If so, is this a WARN or just an INFO?

Reply:
Yeah, INFO makes more sense.